src/os/bluestore/BlueStore.cc

```diff
@@ -5192,21 +5195,22 @@ int BlueStore::_minimal_open_bluefs(bool create)
       bdev->get_size() * (cct->_conf->bluestore_bluefs_min_ratio +
 			  cct->_conf->bluestore_bluefs_gift_ratio);
     initial = std::max(initial, cct->_conf->bluestore_bluefs_min);
-    if (cct->_conf->bluefs_alloc_size % min_alloc_size) {
+    uint64_t bluefs_alloc_size = bluefs->get_bluefs_alloc_size();
+    if (bluefs_alloc_size % min_alloc_size) {
       derr << __func__ << " bluefs_alloc_size 0x" << std::hex
-	    << cct->_conf->bluefs_alloc_size << " is not a multiple of "
+	    << bluefs_alloc_size << " is not a multiple of "
 	    << "min_alloc_size 0x" << min_alloc_size << std::dec << dendl;
       r = -EINVAL;
       goto free_bluefs;
     }
     // align to bluefs's alloc_size
-    initial = p2roundup(initial, cct->_conf->bluefs_alloc_size);
+    initial = p2roundup(initial, bluefs_alloc_size);
     // put bluefs in the middle of the device in case it is an HDD
     uint64_t start = p2align((bdev->get_size() - initial) / 2,
-			      cct->_conf->bluefs_alloc_size);
+			      bluefs_alloc_size);
     //avoiding superblock overwrite
-    ceph_assert(cct->_conf->bluefs_alloc_size > _get_ondisk_reserved());
-    start = std::max(cct->_conf->bluefs_alloc_size, start);
+    ceph_assert(bluefs_alloc_size >= _get_ondisk_reserved());
```

Comment:
I think we can get rid off this assertion (and hence the constraint) and allow 4KB for bluefs_alloc_size. One simply needs to refactor start offset calculation logic a bit to be sure it's never less than _get_ondisk_reserved()....

Reply:
Yeah, this seems like the most straightforward thing...there's no reason why the 2 blocks at the beginning have to fall within a single allocation unit.

Reply:
@liewegas @ifed01 so should we not do https://github.com/ceph/ceph/pull/29384, and just get rid of this assert in this PR? Is there any other extra handling required?

Reply:
@neha-ojha - IMO we can close #29384 and do the following:
1) remove this assert and add new one two line below (after last start assignment):
  assert(start >=_get_ondisk_reserved())
2) (optional but IMO quite useful to protect from wrong user settings)
assert((bluefs_alloc_size % bluestore_min_alloc_size) == 0);
I.e. bluefs_alloc_size >= bluestore_min_alloc_size and aligned with it!
This one can be done at _default_bluefs_alloc_size() when it uses (cct->_conf->bluefs_alloc_size

Reply:
:+1: on both of the above